### PR TITLE
adding Network

### PR DESCRIPTION
**Pull request by jayarjo**

### CODE REVIEW

**Comment by jayarjo** [comment]
Addresses https://github.com/testcontainers/testcontainers-node/issues/66

File: src/network.test.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    expect(exitCode).toBe(0);
+
+    await container1.stop();
+    await container2.stop();
+    await network.close();
+  });
+
+  it("instantiating Network should throw", () => {
+    expect(() => new Network()).toThrowError("use static newNetwork() method to instantiate network");
```
Comment (cristianrgreco) [inline src/network.test.ts:48]: Why not use the constructor instead of having an additional method?

File: src/network.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+class StartedNetwork {
+  constructor(
+    private readonly id: string,
+    private readonly options: CreateNetworkOptions,
+    private readonly dockerClientFactory: DockerodeClientFactory
+  ) {}
+
+  public getId(): string {
+    return this.id!;
```
Comment (cristianrgreco) [inline src/network.ts:13]: Is the `!` required?

File: src/network.test.ts (hunk 1 above)
Comment (jayarjo) [reply src/network.test.ts:48]: That's how it was implemented in `testcontainers-java` - i thought we were meant to take inspiration from there.

File: src/network.ts (hunk 2 above)
Comment (jayarjo) [reply src/network.ts:13]: I wasn't sure how to resolve TypeScript complaint here - the problem is that in the beginning `id` is undefined, so it requests to support that case in typings as well I guess.

**Comment by jayarjo** [comment]
> This looks great, thank you! Could you update the README with some examples on how to use it?

Ok, will do.

**Comment by jayarjo** [comment]
@cristianrgreco  I've added an example to README.  It keeps auto-formatting it into the different form though. Let me know if you expected anything else.

Also in my fork, I have whole lot of [other useful methods](https://github.com/jayarjo/testcontainers-node/blob/dist/src/network.ts) added to the Network. Like ability to grab an existing Network byId or byName, inspect it and check if there are any containers attached to it.

**Comment by cristianrgreco** [comment]
@jayarjo Are you happy for me to merge this PR? 
I will make the two changes I suggested once merged

**Comment by jayarjo** [comment]
@cristianrgreco sure.

**Comment by jayarjo** [comment]
Btw, running tests on every commit is a bit excessive I think, could run on push maybe?

**Pull request merged**